agenta-backend/agenta_backend/services/evaluators_service.py

```diff
@@ -24,23 +29,24 @@ def auto_similarity_match(variant_output, correct_answer, settings_values) -> Re
     return result
 
 
-def auto_regex_test(test_string, regex, should_match):
-    re_pattern = re.compile(regex, re.IGNORECASE)
-    result = bool(re_pattern.search(test_string))
-    return result == should_match
+def auto_regex_test(_, test_string: str, settings_values: dict) -> Result:
+    re_pattern = re.compile(settings_values["regex_pattern"], re.IGNORECASE)
+    result = (
+        bool(re_pattern.search(test_string)) == settings_values["regex_should_match"]
+    )
+    return Result(type="bool", value=result)
```

Comment:
@aakrem I don't understand the usage of this DBModel Result here? What is the advantage?
Why not assume that all evaluators return a float value between 0 and 1 and call it a day? (basically normalized values)

Reply:
It's a generic type to be returned for all evaluators that's it. But better is to use an application model instead of the db model